src/analysis/souffle/analyze_tag_checks_test_helper.dl

```diff
@@ -0,0 +1,74 @@
+#ifndef SRC_ANALYSIS_SOUFFLE_ANALYZE_TAG_CHECKS_TEST_HELPER_DL_
```

Comment:
In this file, I see that you are trying to add some helper relations to write tests, but I am not following everything completely.  I have left some questions at the various definitions. 

In any case, I think we should probably start with something simple like `testFails(message)` (or `testOK` depending upon how we validate success of a test).

Reply:
Done.